spec/Section 3 -- Type System.md

```diff
@@ -2030,28 +2030,52 @@ must *not* be queried if either the `@skip` condition is true *or* the
 ### @deprecated
 
 ```graphql
-directive @deprecated(
-  reason: String = "No longer supported"
-) on FIELD_DEFINITION | ENUM_VALUE
+directive @deprecated(reason: String = "No longer supported") on
+  | FIELD_DEFINITION
+  | ARGUMENT_DEFINITION
+  | INPUT_FIELD_DEFINITION
+  | ENUM_VALUE
 ```
 
 The `@deprecated` directive is used within the type system definition language
 to indicate deprecated portions of a GraphQL service's schema, such as
-deprecated fields on a type or deprecated enum values.
+deprecated fields on a type, arguments on a field, input fields on an input
+type, or values of an enum type.
 
 Deprecations include a reason for why it is deprecated, which is formatted using
 Markdown syntax (as specified by [CommonMark](https://commonmark.org/)).
 
 In this example type definition, `oldField` is deprecated in favor of
-using `newField`.
+using `newField` and `oldArg` is deprecated in favor of using `newArg`.
 
 ```graphql example
 type ExampleType {
   newField: String
   oldField: String @deprecated(reason: "Use `newField`.")
+
+  anotherField(
+    newArg: String
+    oldArg: String @deprecated(reason: "Use `newArg`.")
+  ): String
+}
+```
+
+The `@deprecated` directive must not appear on required (non-null without a default)
+arguments or input object field definitions. Deprecated arguments and fields are 
+excluded by default in introspection, and deprecating required arguments or 
+input fields could create confusion for clients.
```

Comment:
it's not a 'confusion', more like unresolvable conflict. You can't stop using a deprecated arg if it is still required.